frontend/src/components/ibutsu-header.js

```diff
@@ -187,55 +205,154 @@ export class IbutsuHeader extends React.Component {
+        }
+      }
+
+      this.setState({
+        filteredProjects: newSelectOptionsProject,
+      });
+    }
+  }
+
+  toggle = toggleRef => (
+    <MenuToggle
+      variant="typeahead"
+      onClick={this.onProjectToggle}
+      isExpanded={this.state.isProjectSelectorOpen}
+      isFullWidth
+      innerRef={toggleRef}
+    >
+      <TextInputGroup isPlain>
+        <TextInputGroupMain
+          value={this.state.inputValue}
+          onClick={this.onProjectToggle}
+          onChange={this.onTextInputChange}
+          id="typeahead-select-input"
+          autoComplete="off"
+          placeholder="No active project"
+          role="combobox"
+          isExpanded={this.state.isProjectSelectorOpen}
+          aria-controls="select-typeahead-listbox"
+        />
+        <TextInputGroupUtilities>
+          {!!this.state.inputValue && (
+            <Button variant="plain" onClick={() => {
+              this.onProjectClear()
+            }} aria-label="Clear input value">
+              <TimesIcon aria-hidden />
+            </Button>
+          )}
+        </TextInputGroupUtilities>
+      </TextInputGroup>
+    </MenuToggle>
+  );
+
   render() {
     document.title = 'Ibutsu';
     const apiUiUrl = Settings.serverUrl + '/ui/';
     const uploadParams = {};
-    if (this.state.selectedProject && this.state.selectedProject.project) {
-      uploadParams['project'] = this.state.selectedProject.project.id;
+    const {
+      selectedProject,
+      projects,
+      filteredProjects,
+      filterValue,
+    } = this.state;
+
+    if (selectedProject) {
+      uploadParams['project'] = selectedProject.id;
     }
     const topNav = (
       <Flex>
         <FlexItem id="project-selector">
           <Select
-            typeAheadAriaLabel="Select a project"
-            placeholderText="No active project"
-            variant={SelectVariant.typeahead}
+            id="typeahead-select"
             isOpen={this.state.isProjectSelectorOpen}
-            selections={this.state.selectedProject}
-            onToggle={this.onProjectToggle}
+            selected={selectedProject}
             onSelect={this.onProjectSelect}
-            onClear={this.onProjectClear}
-            onTypeaheadInputChanged={this.onProjectsChanged}
-            footer={this.state.projects.length === 10 && "Search for more..."}
+            onOpenChange={() => {
+              this.setState({isProjectSelectorOpen: false});
+            }}
+            toggle={this.toggle}
           >
-            {this.state.projects.map(project => (
-              <SelectOption key={project.id} value={projectToOption(project)} description={project.name} />
-            ))}
+            <SelectList id="select-typeahead-listbox">
+              {(projects.length === 0 && !filterValue) && (
+                <SelectOption
+                  isDisabled={true}
+                  description={"Ask Ibutsu admins to assing you a project"}>
```

Comment:
"Ask Ibutsu admins to add you to a project"

Reply:
Or maybe you were going for "Ask Ibutsu admins to assign you a project", that works too